request/random.go

```diff
@@ -410,6 +415,125 @@ func (b *requestPatchBuilder) Build(cli rest.Interface) Requester {
 	}
 }
 
+type requestPostDelBuilder struct {
+	version         schema.GroupVersion
+	resource        string
+	resourceVersion string
+	namespace       string
+	deleteRatio     float64
+	maxRetries      int
+
+	// Per-builder cache for created resources
+	cache *Cache
+
+	// Per-builder atomic counter for unique ID generation
+	resourceCounter int64
+}
+
+func newRequestPostDelBuilder(src *types.RequestPostDel, resourceVersion string, maxRetries int) *requestPostDelBuilder {
+	return &requestPostDelBuilder{
+		version:         schema.GroupVersion{Group: src.Group, Version: src.Version},
+		resource:        src.Resource,
+		resourceVersion: resourceVersion,
+		namespace:       src.Namespace,
+		deleteRatio:     src.DeleteRatio,
+		maxRetries:      maxRetries,
+		cache:           InitCache(), // Initialize the cache
+	}
+}
+
+// Build implements RequestBuilder.Build.
+func (b *requestPostDelBuilder) Build(cli rest.Interface) Requester {
+	comps := make([]string, 0, 5)
+	if b.version.Group == "" {
+		comps = append(comps, "api", b.version.Version)
+	} else {
+		comps = append(comps, "apis", b.version.Group, b.version.Version)
+	}
+	if b.namespace != "" {
+		comps = append(comps, "namespaces", b.namespace)
+	}
+
+	// Random pick operation DELETE or CREATE based on deleteRatio weight probability
+	randomInt, _ := rand.Int(rand.Reader, big.NewInt(1000))
+	shouldDelete := float64(randomInt.Int64())/1000.0 < b.deleteRatio
+
+	if shouldDelete {
+		// Try to get a name from cache
+		if name, ok := b.cache.Pop(); ok {
+			comps = append(comps, b.resource, name)
+
+			return &PostDelDiscardRequester{
+				builder:   b,
+				name:      name,
+				operation: "DELETE",
+				DiscardRequester: DiscardRequester{
+					BaseRequester: BaseRequester{
+						method: "DELETE",
+						req: cli.Delete().AbsPath(comps...).
+							MaxRetries(b.maxRetries),
+					},
+				},
+			}
+		}
+		// If cache is empty, fall through to POST
+	}
+
+	// POST logic - create resource and add to cache if successful
+	comps = append(comps, b.resource)
+
+	// Use builder's atomic counter for synchronized unique ID generation
+	counter := atomic.AddInt64(&b.resourceCounter, 1)
+	timestamp := time.Now().UnixNano()
+	name := fmt.Sprintf("%s-%s-%d-%d", b.resource, b.namespace, timestamp, counter)
```

Comment:
suggest using `name = fmt.Sprintf("%d-%d", timestamp, counter)` because both resource and namespace are duplicated. you can address it in the follow-up.